src/test/java/com/marklogic/spark/writer/WriteRowsTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.spark.writer;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.DatabaseClientFactory;
+import com.marklogic.client.document.TextDocumentManager;
+import com.marklogic.spark.AbstractIntegrationTest;
+import org.apache.spark.sql.*;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+
+public class WriteRowsTest extends AbstractIntegrationTest {
```

Comment:
This failed for me when I ran it because it's depending on port 8012. Our test app is running on port 8016 though. 

But it's easy to convert this to a real test and get it working - just use the following, which turns `main` into a test method and then uses the config from the parent class to connect to MarkLogic:

```
@Test
    void test() {
        SparkSession sparkSession = newSparkSession();
        Dataset<Row> dataset = sparkSession.read().option("header", true)
            .format("csv")
            .csv("src/test/ml-data/data.csv"); // Path from repository root

        // TODO: The dataset passed in should be available to read (TableProvider or DataWriter?)
        System.out.println(dataset); // [docNum: string, docName: string]
        StringBuffer headers = new StringBuffer();
        for (int i = 0; i < dataset.schema().fields().length; i++) {
            headers.append(dataset.schema().fields()[i].name());
            headers.append(",");
        }
        System.out.println("************ Starting MarkLogicSparkWriteDriver ****************");
        DataFrameWriter<Row> dataFrameWriter = dataset
            .write()
            .format(MarkLogicWriteDataSource.class.getName())
            .option("spark.marklogic.client.host", testConfig.getHost())
            .option("spark.marklogic.client.port", testConfig.getRestPort())
            .option("spark.marklogic.client.username", testConfig.getUsername())
            .option("spark.marklogic.client.password", testConfig.getPassword())
            .option("spark.marklogic.client.authType", "digest")
            // TODO: The below schema passed in should be readable by the inferschema method.
            .option("schema", headers.toString())
            .mode(SaveMode.Append); // mode is needed, else the doc is not written

        dataFrameWriter.save();
        verifyDocIsWritten();
    }

    private void verifyDocIsWritten() {
        TextDocumentManager textDocumentManager = getDatabaseClient().newTextDocumentManager();
        assertNotNull(textDocumentManager.read("doc1.txt"));
    }
```

Reply:
Updated the test, but it throes "NullPointerException" on testConfig.getHost(). Maybe "testConfig" is not populated.